react/core/main.tsx

```diff
@@ -243,13 +244,18 @@ function start() {
       }
 
       if (type === 'img') {
-        props.src = optimizeSrcForVtexImg(vtexImgHost, props.src)
-        if (
-          typeof props.src === 'string' &&
-          props.src.startsWith(vtexImgHost)
-        ) {
+        const { src } = props
+        props.src = optimizeSrcForVtexImg(vtexImgHost, src)
+        if (typeof src === 'string' && src.startsWith(vtexImgHost)) {
```

Comment:
this is not them same code as above.
You are comparing the old reference of `src`, so inside the if, you will be evaluating the value BEFORE the optimizeSrcForVtexImg call. Is this intended?

Reply:
this was not intended, nice catch